siliconcompiler/core.py

```diff
@@ -421,16 +422,25 @@ def create_cmdline(self,
             if not switchlist or any(switch in switchlist for switch in switchstrs):
                 used_switches.update(switchstrs)
                 if typestr == 'bool':
-                    parser.add_argument(*switchstrs,
-                                        nargs='?',
-                                        metavar=metavar,
-                                        dest=dest,
-                                        const='true',
-                                        help=helpstr,
-                                        default=argparse.SUPPRESS)
-                # list type arguments
-                elif re.match(r'\[', typestr):
-                    # all the rest
+                    # Boolean type arguments
+                    if pernodestr == 'never':
+                        parser.add_argument(*switchstrs,
+                                            nargs='?',
+                                            metavar=metavar,
+                                            dest=dest,
+                                            const='true',
+                                            help=helpstr,
+                                            default=argparse.SUPPRESS)
+                    else:
+                        parser.add_argument(*switchstrs,
+                                            metavar=metavar,
+                                            nargs='?',
+                                            dest=dest,
+                                            action='append',
+                                            help=helpstr,
+                                            default=argparse.SUPPRESS)
+                elif re.match(r'\[', typestr) or pernodestr != 'never':
```

Comment:
Why `or` instead of `and`?

Reply:
because both need a list behavior

Reply:
So the typestr would be not starting with `\[` for an option that is set for multiple nodes?

Reply:
`re.match(r'\[', typestr)` this is checking if it's a list of some kind (indicating we need to allow lists), the second is checking if `pernode` is used and if it is, we need to allow multiple values as well

Reply:
Mixed this up as I didn't realise this is still talking about a list in the arguments...Not the list we later get when retrieving an option.